src/fides/api/ops/models/datasetconfig.py

```diff
@@ -117,6 +113,7 @@ def upsert_ctl_dataset(ctl_dataset_obj: Optional[CtlDataset]) -> CtlDataset:
                 fetched_ctl_dataset
             )  # Create/update existing ctl_dataset first
             data["ctl_dataset_id"] = ctl_dataset.id
+            data.pop("dataset", None)
```

Comment:
This method allows both the CTL Dataset and the DatasetConfig to be upserted. The `dataset` is passed in for the CTL Dataset, but we don't want this field when we update the DatasetConfig.